modules/apps/frontend-editor/frontend-editor-ckeditor-web/src/main/resources/META-INF/resources/_diffs/plugins/tabletoolbar/plugin.js

```diff
@@ -0,0 +1,1475 @@
+		const table = tr.getAscendant("table");
+
+		const map = CKEDITOR.tools.buildTableMap(table);
+
+		const rowIndex = tr.$.rowIndex;
+
+		const colIndex = cellInRow(map, rowIndex, cell);
+
+		const rowSpan = cell.$.rowSpan;
+
+		let newCell;
+
+		let newRowSpan;
+
+		let newCellRowSpan;
+
+		let newRowIndex;
+
+		if (rowSpan > 1) {
+			newRowSpan = Math.ceil(rowSpan / 2);
+			newCellRowSpan = Math.floor(rowSpan / 2);
+			newRowIndex = rowIndex + newRowSpan;
+			const newCellTr = new CKEDITOR.dom.element(table.$.rows[newRowIndex]);
+
+			const newCellRow = cellInRow(map, newRowIndex);
+
+			let candidateCell;
+
+			newCell = cell.clone();
+
+			// Figure out where to insert the new cell by checking the vitual row.
+
+			for (let c = 0; c < newCellRow.length; c++) {
+				candidateCell = newCellRow[c];
+
+				// Catch first cell actually following the column.
+
+				if (candidateCell.parentNode == newCellTr.$ && c > colIndex) {
+					newCell.insertBefore(new CKEDITOR.dom.element(candidateCell));
+					break;
+				} else {
+					candidateCell = null;
+				}
+			}
+
+			// The destination row is empty, append at will.
+
+			if (!candidateCell) {
+				newCellTr.append(newCell);
+			}
+		} else {
+			newCellRowSpan = newRowSpan = 1;
+
+			const newCellTr = tr.clone();
+			newCellTr.insertAfter(tr);
+			newCellTr.append((newCell = cell.clone()));
+
+			const cellsInSameRow = cellInRow(map, rowIndex);
+			for (let i = 0; i < cellsInSameRow.length; i++) {
+				cellsInSameRow[i].rowSpan++;
+			}
+		}
+
+		newCell.appendBogus();
+
+		cell.$.rowSpan = newRowSpan;
+		newCell.$.rowSpan = newCellRowSpan;
+		if (newRowSpan == 1) {
+			cell.removeAttribute("rowSpan");
+		}
+		if (newCellRowSpan == 1) {
+			newCell.removeAttribute("rowSpan");
+		}
+
+		return newCell;
+	}
+
+	function horizontalSplitCell(selection, isDetect) {
+		const cells = getSelectedCells(selection);
+		if (cells.length > 1) {
+			return false;
+		} else if (isDetect) {
+			return true;
+		}
+
+		const cell = cells[0];
+
+		const tr = cell.getParent();
+
+		const table = tr.getAscendant("table");
+
+		const map = CKEDITOR.tools.buildTableMap(table);
+
+		const rowIndex = tr.$.rowIndex;
+
+		const colIndex = cellInRow(map, rowIndex, cell);
+
+		const colSpan = cell.$.colSpan;
+
+		let newColSpan;
+
+		let newCellColSpan;
+
+		if (colSpan > 1) {
+			newColSpan = Math.ceil(colSpan / 2);
+			newCellColSpan = Math.floor(colSpan / 2);
+		} else {
+			newCellColSpan = newColSpan = 1;
+			const cellsInSameCol = cellInCol(map, colIndex);
+			for (let i = 0; i < cellsInSameCol.length; i++) {
+				cellsInSameCol[i].colSpan++;
+			}
+		}
+		const newCell = cell.clone();
+		newCell.insertAfter(cell);
+		newCell.appendBogus();
+
+		cell.$.colSpan = newColSpan;
+		newCell.$.colSpan = newCellColSpan;
+		if (newColSpan == 1) {
+			cell.removeAttribute("colSpan");
+		}
+		if (newCellColSpan == 1) {
+			newCell.removeAttribute("colSpan");
+		}
+
+		return newCell;
+	}
+
+	CKEDITOR.plugins.add("tabletoolbar", {
```

Comment:
To be consistent with other plugins, add a duplicate check on top:
```js
(function () {
	const pluginName = 'tabletoolbar';

	if (CKEDITOR.plugins.get(pluginName)) {
		return;
	}

	...

	CKEDITOR.plugins.add(pluginName, {
```

Reply:
Makes sense. Adding! Thanks Marko <3